web/src/add_streams_pill.ts

```diff
@@ -0,0 +1,64 @@
+import * as input_pill from "./input_pill";
+import {set_up} from "./pill_typeahead";
+import * as stream_data from "./stream_data";
+import * as stream_pill from "./stream_pill";
+import type {StreamSubscription} from "./sub_store";
+import type {CombinedPillContainer, CombinedPillItem} from "./typeahead_helper";
+
+type SetUpPillTypeaheadConfig = {
+    pill_widget: CombinedPillContainer;
+    $pill_container: JQuery;
+};
+
+function display_pill(sub: StreamSubscription): string {
+    return "#" + sub.name;
+}
+
+function create_item_from_text(
+    stream_name: string,
+    current_items: CombinedPillItem[],
+): input_pill.InputPillItem<stream_pill.StreamPill> | undefined {
+    stream_name = "#" + stream_name;
+    return stream_pill.create_item_from_stream_name(stream_name, current_items, display_pill);
+}
+
+function get_text_from_item(item: CombinedPillItem): string {
+    let text = "";
+    if (item.type === "stream") {
+        text = stream_pill.get_stream_name_from_item(item);
+    }
+    return text;
+}
+
+function set_up_pill_typeahead({pill_widget, $pill_container}: SetUpPillTypeaheadConfig): void {
+    const opts = {
+        stream: true,
+        user_group: false,
+        user: false,
+        exclude_bots: true,
+        help_on_empty_strings: true,
+        display_pill,
+    };
+    set_up($pill_container.find(".input"), pill_widget, opts);
+}
+
+export function add_default_stream_pills(pill_widget: CombinedPillContainer): void {
+    const default_stream_ids = stream_data.get_default_stream_ids();
+    for (const stream_id of default_stream_ids) {
+        const sub = stream_data.get_sub_by_id(stream_id);
+        if (sub) {
+            stream_pill.append_stream(sub, pill_widget, display_pill);
+        }
+    }
+}
+
+export function create($stream_pill_container: JQuery): CombinedPillContainer {
+    const pill_widget = input_pill.create<CombinedPillItem>({
+        $container: $stream_pill_container,
+        create_item_from_text,
+        get_text_from_item,
+    });
+    add_default_stream_pills(pill_widget);
+    set_up_pill_typeahead({pill_widget, $pill_container: $stream_pill_container});
+    return pill_widget;
+}
```

Comment:
How did you pick the name for this module? I think the name is too generic, since it's not a generic streams picker, but instead one specifically designed for the invitations stream picker; a good name would reflect that, like `invite_stream_picker_pill.ts`

Reply:
Yeah I am pretty bad at naming files and functions.

Reply:
It's a hard skill! I think the key to improving is to try spend a few minutes thinking about each larger thing (function/module/etc.) you name, and then you can check your reasoning by including a comment in your PRs next to each name you introduce explaining how you picked it.

Reply:
Thank you, I'll try this.